### PR TITLE
Catalogs functional test

### PR DESCRIPTION
**Pull request by srinia6**
@dalebremner @brianparry @geoff-reid @abildw @RackHD/corecommitters

### CODE REVIEW

**Comment by srinia6** [comment]
test this please

**Comment by srinia6** [comment]
test this please

**Comment by srinia6** [comment]
test this please

**Comment by geoff-reid** [comment]
:+1:

**Comment by zyoung51** [comment]
This exercises the APIs ability to route to a handler, but doesn't ensure that expected data is being returned.  Will this eventually functionally validate that the compute node has dmi, ohai, smart, etc catalogs?  

+1 for exercising the route and as a starting point for performing functional validation later.

**Comment by srinia6** [comment]
@zyoung51 - I can add more tests to validate returned catalog data going forward.

File: test/tests/api/v2_0/catalogs_tests.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class CatalogsTests(object):
+    def __init__(self):
+        self.__client = config.api_client
+
+    @test(groups=['catalogs_api2.tests', 'api2_check-catalogs'])
+    def test_catalogs(self):
+        """Testing GET:api/2.0/catalogs to get list of catalogs"""
+        Api().get_catalog()
+        rsp = self.__client.last_response
+        assert_not_equal(404, rsp.status, message=rsp.reason)
```
Comment (brianparry) [inline test/tests/api/v2_0/catalogs_tests.py]: @srinia6 This check seems unnecessary.  I think the `assert_equal(200, rsp.status, ...)` check is sufficient by itself.

**Comment by brianparry** [comment]
:+1: One minor comment that shouldn't test results.  I'd suggest adding a story to the backlog to expand these tests as suggested by @zyoung51

**Comment by brianparry** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Pull request merged**